test/typescript/sample-react-redux-usage.tsx

```diff
@@ -0,0 +1,77 @@
+// This file exists to test our types against sample user code
+// This is compiled using `tsc` in our `test-ts-usage` script
+import React from 'react';
+import ReactDOM from 'react-dom';
+// useSelector
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+
+import equal from '../../index.js';
+
+type IItem = {
+  text: string;
+  id: string;
+};
+
+const testArr: IItem[] = [
+  { text: 'green', id: '23' },
+  { text: 'sunshine', id: '1' },
+  { text: 'mountain', id: '11' },
+  { text: 'air', id: '8' },
+  { text: 'plants', id: '9' },
+  { text: 'air', id: '8' },
+  { text: 'air', id: '8' },
+];
+
+type IChildProps = {
+  item: IItem;
+};
+
+class TestChild extends React.Component<IChildProps> {
```

Comment:
we can totally remove the child from this test. _all_ we need is really state, provider, and a component that inherits from the provide and uses `useSelector` _with react-fast-compare_.